test/e2e/common/downward_api.go

```diff
@@ -104,6 +104,61 @@ var _ = ginkgo.Describe("[sig-node] Downward API", func() {
 		testDownwardAPI(f, podName, env, expectations)
 	})
 
+	/*
+	   Release : v1.16
+	   Testname: DownwardAPI, environment for host ip and pod ip with host network
+	   Description: Downward API MUST expose Pod and Container fields as environment variables. Specify host IP and pod IP as environment variable in the Pod Spec are visible at runtime in the container if the pod uses host network. The host IP and the pod IP has to be the same
+	*/
+	framework.ConformanceIt("should provide host IP and pod IP as an env var if pod uses host network [NodeConformance]", func() {
+		podName := "downward-api-" + string(uuid.NewUUID())
+		env := []v1.EnvVar{
+			{
+				Name: "HOST_IP",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "status.hostIP",
+					},
+				},
+			},
+			{
+				Name: "POD_IP",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "status.podIP",
+					},
+				},
+			},
+		}
+
+		expectations := []string{
+			fmt.Sprintf("OK"),
+		}
+
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"name": podName},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "dapi-container",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"sh", "-c", "if [[ $HOST_IP == $POD_IP ]]; then echo OK; fi"},
```

Comment:
@BenTheElder what´s the right way to check both are identical using busybox shell?

Reply:
this should work but:
```suggestion
						Command: []string{"sh", "-c", `[[ "${HOST_IP}" == "${POD_IP}" ]] && echo 'OK'`},
```
This only strictly checks that they are equal though, it will also echo `OK` if they are both unset.

This version requires both to be set:
```suggestion
						Command: []string{"sh", "-c", `[[ "${HOST_IP:?}" == "${POD_IP:?}" ]] && echo 'OK'`},
```

local output (unset for both):
```
/ # [[ "${HOST_IP}" == "${POD_IP}" ]] && echo 'OK'
OK
/ # [[ "${HOST_IP:?}" == "${POD_IP:?}" ]] && echo 'OK'
sh: HOST_IP: parameter not set or null
```

Reply:
nice to have you close 🥇